openmc/data/resonance_covariance.py

```diff
@@ -0,0 +1,841 @@
+                         #uncertainties followed by compact correlations
+            items, values = get_list_record(file_obj)
+            mean = items
+            num_res = items[5] 
+            energy = values[0::12]
+            spin = values[1::12]
+            gt = values[2::12]
+            gn = values[3::12]
+            gg = values[4::12]
+            gf = values[5::12]
+            par_unc = []
+            for i in range(num_res):
+                res_unc = values[i*12+6:i*12+12]
+                #Delete 0 values (not provided, no fission width)
+                # DAJ/DGT always zero, DGF sometimes none zero [1,2,5]
+                res_unc_nonzero = []
+                for j in range(6):
+                    if j in [1,2,5] and res_unc[j] != 0.0 :
+                        res_unc_nonzero.append(res_unc[j])
+                    elif j in [0,3,4]:
+                        res_unc_nonzero.append(res_unc[j])
+                par_unc.extend(res_unc_nonzero)
+
+            records = []
+            for i, E in enumerate(energy):
+                records.append([energy[i], spin[i], gt[i], gn[i],
+                                gg[i], gf[i]])
+
+            corr = get_intg_record(file_obj)
+            cov = np.diag(par_unc).dot(corr).dot(np.diag(par_unc))
+
+            # Create pandas DataFrame with resonacne data
+            columns = ['energy', 'J', 'totalWidth', 'neutronWidth',
+                       'captureWidth', 'fissionWidth']
+            parameters = pd.DataFrame.from_records(records, columns=columns)
+
+            #Determine mpar (number of parameters for each resonance in
+            #covariance matrix)
+            nparams,params = parameters.shape
+            covsize = cov.shape[0]
+            mpar = int(covsize/nparams)
+            
+            #Add parameters from File 2
+            parameters = file2contributions(parameters, file2params)
+
+            # Create instance of MultiLevelBreitWignerCovariance
+            mlbw = cls(energy_min, energy_max)
+            mlbw.parameters = parameters
+            mlbw.covariance = cov
+            mlbw.mpar  = mpar
+            mlbw.lcomp = LCOMP
+
+            return mlbw
+
+        elif LCOMP == 0 :
+            cov = np.zeros([4,4])
+            records = []
+            cov_index = 0
+            for i in range(NLS):
+                items, values = get_list_record(file_obj)
+                num_res = items[5]
+                for j in range(num_res):
+                    one_res = values[18*j:18*(j+1)]
+                    res_values = one_res[:6]
+                    cov_values = one_res[6:]
+
+                    energy = res_values[0]
+                    spin = res_values[1]
+                    gt = res_values[2]
+                    gn = res_values[3]
+                    gg = res_values[4]
+                    gf = res_values[5]
+                    records.append([energy, spin, gt, gn, gg, gf])
```

Comment:
Looks like you could just: `records.append(list(res_values))`. No need to assign energy, spin, etc.